pkg/cmd/task/wait/wait.go

```diff
@@ -52,20 +57,26 @@ func NewCmdWait(f factory.Factory) *cobra.Command {
 			dependencies := cmd.NewDependencies(f, c)
 			opts := NewWaitOps(dependencies, taskIDs)
 
-			return WaitRun(opts.Out, taskIDs, opts.GetServerTasksCallback, timeout)
+			return WaitRun(opts.Out, taskIDs, opts.GetServerTasksCallback, opts.GetTaskDetailsCallback, timeout, showProgress)
 		},
 	}
 
 	flags := cmd.Flags()
 	flags.IntVar(&timeout, FlagTimeout, DefaultTimeout, "Duration to wait (in seconds) before stopping execution")
+	flags.BoolVar(&showProgress, FlagProgress, false, "Show detailed progress of the tasks")
 
 	return cmd
 }
 
-func WaitRun(out io.Writer, taskIDs []string, getServerTasksCallback ServerTasksCallback, timeout int) error {
+func WaitRun(out io.Writer, taskIDs []string, getServerTasksCallback ServerTasksCallback, getTaskDetailsCallback TaskDetailsCallback, timeout int, showProgress bool) error {
```

Comment:
This could be reduced by just passing in the `WaitOptions` struct rather than all of its properties.